src/server/middleware/immutable/deletemf.go

```diff
@@ -43,46 +40,21 @@ func handleDelete(req *http.Request) error {
 		return errors.New("cannot get the manifest information from request context")
 	}
 
-	_, repoName := common_util.ParseRepository(mf.Repository)
-	total, repos, err := repository.Mgr.List(req.Context(), &q.Query{
-		Keywords: map[string]interface{}{
-			"Name": mf.Repository,
-		},
-	})
-	if err != nil {
-		return err
-	}
-	if total == 0 {
-		return nil
-	}
-
-	total, afs, err := artifact.Mgr.List(req.Context(), &q.Query{
-		Keywords: map[string]interface{}{
-			"ProjectID":    mf.ProjectID,
-			"RepositoryID": repos[0].RepositoryID,
-			"Digest":       mf.Digest,
-		},
-	})
-	if err != nil {
-		return err
-	}
-	if total == 0 {
-		return nil
-	}
-
-	total, tags, err := tag.Mgr.List(req.Context(), &q.Query{
-		Keywords: map[string]interface{}{
-			"ArtifactID": afs[0].ID,
-		},
+	af, err := artifact.Ctl.GetByReference(req.Context(), mf.Repository, mf.Digest, &artifact.Option{
+		WithTag: true,
 	})
 	if err != nil {
+		if internal_errors.IsErr(err, internal_errors.NotFoundCode) {
+			return nil
+		}
 		return err
 	}
-	if total == 0 {
+	if af == nil {
 		return nil
 	}
 
-	for _, tag := range tags {
+	_, repoName := common_util.ParseRepository(mf.Repository)
+	for _, tag := range af.Tags {
```

Comment:
Set with immutable as true, the immutable of tags status will be returned back. So we don't need to do the match again

Reply:
updated.